### PR TITLE
Adds code folding and matching html tag highlighting

### PR DESCRIPTION
**Pull request by timmaffett**
This PR adds the codemirror options and script includes to activate code folding.
This PR also adds matching HTML tag highlighting.

Examples of the code folding UI added to the current master can be seen here:

dark theme:
https://timmaffett.github.io/folding/?id=91d894c115314fbe2fb32e4adb91d17f
light theme:
https://timmaffett.github.io/folding_minified/embed-html.html?id=91d894c115314fbe2fb32e4adb91d17f

(the first example uses codemirror and includes as they are in this PR/current pub dev, and the light theme example uses the codemirror minified with PR 150 (see below) and the html uses only the 2 codemirror includes as shown below).

Note:
If codemirror.dart is updated to include PR https://github.com/google/codemirror.dart/pull/150 then all of the html files within dart-pad  (and the added codemirror includes in this PR will also no longer be necessary) will have their included codemirror css/scripts reduced to simply: 
```
<link href="packages/codemirror/css/codemirror.css" rel="stylesheet" media="screen">
<script src="packages/codemirror/codemirror.js" defer></script>
```

### CODE REVIEW

**Comment by domesticmouse** [comment]
Conflicts in need of resolving

**Comment by timmaffett** [comment]
resolved.

File: web/embed-flutter.html [hunk 1]
```diff
@@ -30,6 +30,13 @@
 
     <script src="packages/codemirror/codemirror.js" defer></script>
     <script src="packages/codemirror/addon/scroll/simplescrollbars.js" defer></script>
+    <script src="packages/codemirror/addon/fold/foldcode.js" defer></script>
+    <script src="packages/codemirror/addon/fold/foldgutter.js" defer></script>
+    <script src="packages/codemirror/addon/fold/brace-fold.js" defer></script>
+    <script src="packages/codemirror/addon/fold/xml-fold.js" defer></script>
+    <script src="packages/codemirror/addon/fold/indent-fold.js" defer></script>
+    <script src="packages/codemirror/addon/fold/comment-fold.js" defer></script>
+    <script src="packages/codemirror/addon/edit/matchtags.js" defer></script>
```
Comment (johnpryan) [inline web/embed-flutter.html:39]: Are these also included in codemirror after https://github.com/dart-lang/dart-pad/pull/2214?

Comment (timmaffett) [reply web/embed-flutter.html:39]: Yes they are included in the new codemirror.  So those script includes are no longer required and should now be removed.

Comment (timmaffett) [reply web/embed-flutter.html:39]: (from all of the html files)

Comment (timmaffett) [reply web/embed-flutter.html]: i resolved the conflicts taking the new codemirror ane now merged include file changes into account.

**Review by johnpryan — APPROVED** [review]

**Pull request merged**